src/errors.rs

```diff
@@ -303,9 +303,9 @@ error_chain! {
             description("toolchain does not support components")
             display("toolchain '{}' does not support components", t)
         }
-        UnknownComponent(t: String, c: String) {
+        UnknownComponent(t: String, c: String, s: String) {
             description("toolchain does not contain component")
-            display("toolchain '{}' does not contain component {}", t, c)
+            display("toolchain '{}' does not contain component {}{}?", t, c, s)
```

Comment:
If the `s` parameter is `""` then there'll be a spurious `?` on the string.  Better for `s` to be an `Option<String>` and format accordingly.